magento/pwa-studio
# [bugfix]: search trigger focus

## Description
Clicking the search trigger should focus the search input. This was done before, but we broke it at some point in our talon refactor. I also replaced some components' use of `location` and `history` with the equivalent hooks.

Old issue: https://github.com/magento/pwa-studio/pull/1019

## Current Behavior

Clicked search trigger, press tab, then press shift tab.

[![Image from Gyazo](https://i.gyazo.com/6d914759537d0b19ae1d04059000fe8c.gif)](https://gyazo.com/6d914759537d0b19ae1d04059000fe8c)

## New Behavior

Clicked search trigger, press tab, then shift tab, shift tab.
[![Image from Gyazo](https://i.gyazo.com/795d7e87dc6e6b101dc7ff73d72dc802.gif)](https://gyazo.com/795d7e87dc6e6b101dc7ff73d72dc802)

## Tab Order

I opted not to change tab order, only to focus the input element on click.

This does result in a strange behavior where tabbing out of the search input moves to the first item in the page, thus skipping over the cart trigger.

That said, in my opinion this is a better UX than before.

* Sighted users can now click the trigger and immediately start typing a search term.
* Non sighted users/keyboard-only users are automatically moved to the search input instead of having to tab further to find the field.

## Related Issue





Closes [PWA-371](https://jira.corp.magento.com/browse/PWA-371).

## Acceptance 



### Verification Stakeholders

@soumya-ashok or @schensley and @jimbo 
### Specification



### Verification Steps


1. Open any venia page and click the search trigger.
2. Observe the focus moves directly to the input.
3. Click the search trigger again. Focus should now be back on the search trigger.

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.
